server/app/middlewares/websocket_backend.rb

```diff
@@ -168,26 +172,40 @@ def handle_rpc_notification(ws, data)
     end
   end
 
-  ##
-  # On websocket connection close
+  # Unplug client on websocket connection close.
+  #
+  # The client may have already been unplugged, if we closed the connection.
   #
   # @param [Faye::WebSocket] ws
   def on_close(ws)
     client = @clients.find{|c| c[:ws] == ws}
     if client
-      node = HostNode.find_by(node_id: client[:id])
-      if node
-        Agent::NodeUnplugger.new(node).unplug!
-        logger.info "node closed connection: #{node.name || node.node_id}"
-      end
-      @clients.delete(client)
+      logger.info "node #{client[:id]} connection closed"
+      unplug_client(client)
+    else
+      logger.debug "ignore close of unplugged client"
     end
```

Comment:
This used to call `ws.close` because the ping timeout handler called `on_close` to disconnect the node. Those now call `close_client`, so now this is only called from the `ws.on :close`.

I don't think it makes any sense to close the websocket from the websocket close handler. I guess `Faye::Websocket` just used to ignore it in that case.